modules/adserveradtargetBidAdapter.js

```diff
@@ -0,0 +1,408 @@
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+      case 'objectAllKeys':
+        if (isPlainObject(inputObject)) {
+          outputObject[objectKey] = inputObject[objectKey];
+        };
+        break;
+    };
+  };
+  return outputObject;
+};
+
+function getTtl(bidderRequest) {
+  const ttl = config.getConfig('adserveradtarget.ttl');
+  return ttl ? validateTTL(ttl) : validateTTL(deepAccess(bidderRequest, 'params.ttl'));
+};
+
+function validateTTL(ttl) {
+  return (isNumber(ttl) && ttl > 0 && ttl < 3600) ? ttl : DEFAULT_BID_TTL
+};
+
+function getFloorModuleData(bid) {
+  const getFloorRequestObject = {
+    currency: deepAccess(bid, 'params.bidOverride.cur') || DEFAULT_CURRENCY,
+    mediaType: BANNER,
+    size: '*'
+  };
+  return (isFn(bid.getFloor)) ? bid.getFloor(getFloorRequestObject) : false;
+};
+
+function generateOpenRtbObject(bidderRequest, bid) {
+  if (bidderRequest) {
+    let outBoundBidRequest = {
+      id: generateUUID(),
+      cur: [getFloorModuleData(bidderRequest).currency || deepAccess(bid, 'params.bidOverride.cur') || DEFAULT_CURRENCY],
+      imp: [],
+      site: {
+        page: deepAccess(bidderRequest, 'refererInfo.page')
+      },
+      device: {
+        dnt: 0,
+        ua: navigator.userAgent,
+        ip: deepAccess(bid, 'params.bidOverride.device.ip') || deepAccess(bid, 'params.ext.ip') || undefined
+      },
+      regs: {
+        ext: {
+          'us_privacy': bidderRequest.uspConsent ? bidderRequest.uspConsent : '',
+          gdpr: bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies ? 1 : 0
+        }
+      },
+      source: {
+        ext: {
+          hb: 1,
+          adapterver: ADAPTER_VERSION,
+          prebidver: PREBID_VERSION,
+          integration: {
+            name: INTEGRATION_METHOD,
+            ver: PREBID_VERSION
+          }
+        },
+        fd: 1
+      },
+      user: {
+        ext: {
+          consent: bidderRequest.gdprConsent && bidderRequest.gdprConsent.gdprApplies
+            ? bidderRequest.gdprConsent.consentString : ''
+        }
+      }
+    };
+
+    outBoundBidRequest.site.id = bid.params.sid;
+
+    if (bidderRequest.ortb2) {
+      outBoundBidRequest = appendFirstPartyData(outBoundBidRequest, bid);
+    };
+
+    if (deepAccess(bid, 'schain')) {
+      outBoundBidRequest.source.ext.schain = bid.schain;
+      outBoundBidRequest.source.ext.schain.nodes[0].rid = outBoundBidRequest.id;
+    };
+
+    return outBoundBidRequest;
+  };
+};
+
+function appendImpObject(bid, openRtbObject) {
+  const mediaTypeMode = getMediaType(bid);
+
+  if (openRtbObject && bid) {
+    const impObject = {
+      id: bid.bidId,
+      secure: isSecure(bid),
+      bidfloor: getFloorModuleData(bid).floor || deepAccess(bid, 'params.bidOverride.imp.bidfloor') || 0.000001
+    };
+
+    if (bid.mediaTypes.banner && (typeof mediaTypeMode === 'undefined' || mediaTypeMode === BANNER)) {
```

Comment:
I believe `mediaTypeMode === BANNER` be sufficient here. Your `getMediaType` method return `'banner'` when `bid.mediaTypes.banner` exists and `false` otherwise. I don't see how this `undefined` use case is relevant here

Reply:
fixed